go/x509util/x509util.go

```diff
@@ -0,0 +1,576 @@
+// Package x509util includes utility code for working with X.509
+// certificates from the x509 package.
+package x509util
+
```

Comment:
Are there tests somewhere for this package?  Coming in a future PR is fine, but tests are handy!